### ISSUE TITLE
Auto scroll can be distracting

### ISSUE THREAD

**Issue by bbondy** [brave/brave-browser#32970]
We need to stop auto-scroll as content floods in because you are already reading the top of the content but then can’t as it scrolls out of sight.
I do think we should scroll the top of the answer into view

**Comment by Brave-Matt** [comment]
Agreed — made a similar comment here: https://bravesoftware.slack.com/archives/C04KPE0BUDU/p1698346929744069 

I think we should make it so that if you scroll up at least once, it will _stop automatically scrolling_ down with the response.

**Comment by bbondy** [comment]
I think the solution here is to auto scroll only if the scroll position is at the bottom.

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.63.124 Chromium: 121.0.6167.57 (Official Build) nightly (64-bit)
-- | --
Revision | e0864ffae4be8015ad75c2b344ab75c2eefb31a7
OS | Windows 10 Version 22H2 (Build 19045.3930)
```

### Steps:
1. installed `1.63.124`
2. launched Brave using `--env-leo=staging --env-ai-chat.bsg=dev --env-ai-chat-premium.bsg=dev --enable-logging=stderr`
3. loaded `https://www.masshist.org/thomasjeffersonpapers/doc?id=declaration_1`
4. clicked on the `Leo` icon in the sidebar
5. clicked `Summarize this page`
6. clicked `Accept and begin`
7. scrolled up
8. confirmed auto-scrolling stopped
9. scrolled down
10. confirmed summarized text
11. scrolled down
12. clicked on `Suggest questions...`
13. immediately scrolled up

### Confirmed auto-scrolling text only happens when the view is at the bottom of the content

example | example | example | example | example
----------|----------|-----------|-----------|---------
![image](https://github.com/brave/brave-browser/assets/387249/c4cc799f-d2c1-4293-ae15-e68d691fb169) | ![image](https://github.com/brave/brave-browser/assets/387249/26e1536c-3176-40a9-9e67-c1f3d13ab221) | ![image](https://github.com/brave/brave-browser/assets/387249/5ffbc8c6-60f4-483c-97e3-14294aa1ad34) | ![image](https://github.com/brave/brave-browser/assets/387249/83fb856f-445b-4957-b954-b58d7157a44d) | ![image](https://github.com/brave/brave-browser/assets/387249/db72916d-6a27-4a54-b60f-0225ec0ec38f)

**Comment by kjozwiak** [comment]
Adding `QA/Blocked` for now as the above was merged into `cr121-1.62.x` via https://github.com/brave/brave-core/pull/21593. Once `C121` is merged into `1.62.x` via https://github.com/brave/brave-core/pull/21593, we'll unblock the above and mention the needed `1.62.x` version for verification 👍

**Comment by kjozwiak** [comment]
The above requires `1.62.148` or higher for `1.62.x` verification 👍

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.62.149 Chromium: 121.0.6167.75 (Official Build) (x86_64)
-- | --
Revision | ff84587bd70af9fcbcbe59fc5194ca65082759c4
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Steps:
1. installed `1.62.149`
2. launched Brave using `--env-leo=staging --env-ai-chat.bsg=dev --env-ai-chat-premium.bsg=dev --enable-logging=stderr --v=2`
3. loaded `https://constitutioncenter.org/the-constitution/full-text`
4. clicked on `Summarize this page`
5. waited for auto-scrolling to start
6. swiped up and waited
7. scrolled down

![stop-autoscroll](https://github.com/brave/brave-browser/assets/387249/dd66b737-e856-467e-81ed-cf1174466b47)
